workflow-gen/Activities/ReserveInventoryActivity.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace WorkflowGen.Activities
+{
+    using System.Threading.Tasks;
+    using Dapr.Client;
+    using Dapr.Workflow;
+    using Microsoft.Extensions.Logging;
+    using WorkflowGen.Models;
+    using System;
+
+    class ReserveInventoryActivity : WorkflowActivity<InventoryRequest, InventoryResult>
+    {
+        readonly ILogger logger;
+        readonly DaprClient client;
+        static readonly string storeName = "statestore";
+
+        public ReserveInventoryActivity(ILoggerFactory loggerFactory, DaprClient client)
+        {
+            this.logger = loggerFactory.CreateLogger<ReserveInventoryActivity>();
+            this.client = client;
+        }
+
+        public override async Task<InventoryResult> RunAsync(WorkflowActivityContext context, InventoryRequest req)
+        {
+            this.logger.LogInformation(
+                "Reserving inventory for order {requestId} of {quantity} {name}",
+                req.RequestId,
+                req.Quantity,
+                req.ItemName);
+
+            OrderPayload orderResponse;
+            string key;
+
+            (orderResponse, key) = await client.GetStateAndETagAsync<OrderPayload>(storeName, req.ItemName);
```

Comment:
Nit: You should be able to combine the declaration and the assignment here. You should also be able to use `var`.

Also, the 2nd return value here isn't the key, but rather the ETag. Are we validating it anywhere?